cli/reactjs_jsx_ppx.ml

```diff
@@ -2459,12 +2443,20 @@ module V4 = struct
       match List.filter hasAttr pval_attributes with
       | [] -> [item]
       | [_] ->
+        let hasForwardRef = ref false in
         let rec getPropTypes types ({ptyp_loc; ptyp_desc} as fullType) =
           match ptyp_desc with
           | Ptyp_arrow (name, type_, ({ptyp_desc = Ptyp_arrow _} as rest))
             when isOptional name || isLabelled name ->
             getPropTypes ((name, ptyp_loc, type_) :: types) rest
-          | Ptyp_arrow (Nolabel, _type, rest) -> getPropTypes types rest
+          | Ptyp_arrow
+              ( Nolabel,
+                {ptyp_desc = Ptyp_constr ({txt = Lident "unit"}, _)},
+                rest ) ->
+            getPropTypes types rest
+          | Ptyp_arrow (Nolabel, _type, rest) ->
+            hasForwardRef := true;
```

Comment:
Not sure I understand this.
How does one know that there is a forwardRef just because there's a function with a non-labelled arg which is not unit?

Is this check applying on the user code, or on code generated by the ppx?

Reply:
Good catch. And it is quite a tricky part, because of the usage of forwardRef.
Basically, we are not allowing the nolabel arg in `@react.component`, right? There is one exception to this rule, forwardRef. So, for the interface, it has no way to know what the label is for the type signature. I think we need some assumptions and restrictions at some point. If there is a nolabel type signature after labeled args, the only ref can go there. It will be considered as forwardRef case.

Reply:
When operating on a forwardRef, do we know that it's a forwardRef to begin with?
If that code was gated by a check "we are inside a forwardRef" then I would have no objection to it.
As it would be clear it's just a simple heuristic, and we don't care because it only applies to forwardRef.

Reply:
So I imagine a reference `insideForwardRef` which is set. And checked at this point in the code.